src/features/posts/posts.ts

```diff
@@ -0,0 +1,52 @@
+import { PayloadAction, createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Post } from '../../types/Post';
+import { getUserPosts } from '../../api/posts';
+
+const initialState = {
+  posts: [] as Post[],
+  loaded: false,
+  hasError: false,
+};
+
+export const fetchUserPosts = createAsyncThunk(
+  'posts/fetchUserPosts',
+  async (userId: number) => {
+    const posts = await getUserPosts(userId);
+
+    return posts;
+  },
```

Comment:
```suggestion
  'posts/fetchUserPosts',
  getUserPosts,
```
we can simplify functions in such cases